pkg/devfile/adapters/kubernetes/utils/utils.go

```diff
@@ -78,8 +78,14 @@ func GetContainers(devfileObj devfileParser.DevfileObj) ([]corev1.Container, err
 		for _, c := range containers {
 			for _, containerPort := range c.Ports {
 				for _, curPort := range container.Ports {
+					if curPort.Name == containerPort.Name {
+						// the name has to be unique across containers since it is considered as the URL name
+						return nil, fmt.Errorf("Devfile contains multiple endpoint entries with same name: %v", containerPort.Name)
+					}
```

Comment:
Do we need this? ConvertPorts() seems to already do this check and it is called above before the loop.

Reply:
As described in the comments, ConvertPorts() only checks for duplicate endpoint name within same container. and the check here checks for duplication across containers.

Reply:
Sorry didn't notice the comment